service/internal/workflow/workflow.go

```diff
@@ -21,7 +22,7 @@ type Workflow interface {
 	Logs(ctx context.Context, workflowName string) (*Logs, error)
 	LogStream(ctx context.Context, workflowName string, data http.ResponseWriter) error
 	Status(ctx context.Context, workflowName string) (*Status, error)
-	Submit(ctx context.Context, from string, parameters map[string]string) (string, error)
+	Submit(ctx context.Context, from string, parameters map[string]string, txID string) (string, error)
```

Comment:
I think this should probably be more generic; `labels map[string]string` (similar to `parameters`)?

Reply:
good call, I like that